src/udp_wrap.cc

```diff
@@ -324,6 +328,42 @@ void UDPWrap::DropMembership(const FunctionCallbackInfo<Value>& args) {
   SetMembership(args, UV_LEAVE_GROUP);
 }
 
+void UDPWrap::SetSourceMembership(const FunctionCallbackInfo<Value>& args,
+                                  uv_membership membership) {
+  UDPWrap* wrap;
+  ASSIGN_OR_RETURN_UNWRAP(&wrap,
+                          args.Holder(),
+                          args.GetReturnValue().Set(UV_EBADF));
+
+  CHECK_EQ(args.Length(), 3);
+
+  node::Utf8Value sourceAddress(args.GetIsolate(), args[0]);
+  node::Utf8Value groupAddress(args.GetIsolate(), args[1]);
+
+  const char* iface_cstr = nullptr;
+  if (!args[2]->IsUndefined() && !args[2]->IsNull()) {
+    iface_cstr = *(node::Utf8Value(args.GetIsolate(), args[2]));
```

Comment:
This is going to crash at some point – you need to keep the `Utf8Value` instance alive while `iface_cstr` is used. You’ll need to declare this `Utf8Value` like the others, then you can use `iface_cstr = *iface_utf8value`

Reply:
@addaleax That's how it used to be coded but it was changed to address an earlier review comment by @cjihrig. 

> You can also move node::Utf8Value iface(args.GetIsolate(), args[2]); from above into the else branch so we don't compute it for no reason.

So I have conflicted advice and not sure how to resolve the issue.

Reply:
@LPardue my comment was because you were computing the variable, even when you didn't need it. In other words, `node::Utf8Value iface(args.GetIsolate(), args[2]);` was being executed no matter what, but the value was only used in one path.

I believe @addaleax is saying that you need to assign the value to a variable in the path where you do use it.

Reply:
@cjihrig The thing is, this is really hard to write in a way that avoids the unnecessary computation for the undefined/null cases and is still somewhat readable… The best alternatives would probably be either using a `std::string` for memory management, or dynamically allocating the `Utf8Value` with `new`; both of these come with some runtime overhead as well.

Reply:
@cjihrig I appreciate the issue but struggle to construct the appropriate node::Utf8Value in correct scope. The simple answer would be to branch and call `uv_udp_set_source_membership` in either branch (one with a populated `const char*` the other with `nullptr`). However, duplication is a code smell to me.

Reply:
Unless this should be expected to be very hot code, I’d just go back to the previous state. (I am certainly not a domain expert, but it doesn’t sound like you’d ever want to call this method more than a handful of times in any Node process…)

Reply:
OK, yea, go with what you had before. Sorry for the noise.

Reply:
My anticipated usage is to join SSM sessions at the start of a program and leave them at termination. In certain cases, join/leave requests may be made at a rate of up to 1 every 8 seconds. Other people's use cases may differ.

Thanks for the discussion, I'll revert back to original code.

Reply:
Yeah, I think this is something that’s more on the order of 1 µs or so :smile:

Reply:
@addaleax I believe I've addressed your review comment by implementing the change.